lib/http/request/body.rb

```diff
@@ -55,6 +48,17 @@ def validate_body_type!
 
         raise RequestError, "body of wrong type: #{@body.class}"
       end
+
+      class BlockIO
```

Comment:
Probably worth name it `ProcIO`. It took a while for me to understand that *block* here refers to ruby block, rather than to *blocking*.